Sources/PackageRegistry/RegistryClient.swift

```diff
@@ -22,10 +22,10 @@ import struct TSCUtility.ZipArchiver
 
 /// Package registry client.
 /// API specification: https://github.com/apple/swift-package-manager/blob/main/Documentation/Registry.md
-public enum RegistryError: Error {
+public enum RegistryError: Error, CustomStringConvertible {
     case registryNotConfigured(scope: PackageIdentity.Scope?)
     case invalidPackage(PackageIdentity)
-    case invalidURL
+    case invalidURL(URL)
```

Comment:
It's a little strange to say it's invalid when there is an actual `URL`.

Reply:
do you prefer to make this a string? we always have a URL in all the cases this is thrown